data/battle_scripts_1.s

```diff
@@ -369,6 +369,196 @@ gBattleScriptsForMoveEffects:: @ 82D86A8
 	.4byte BattleScript_EffectSleepHit
 	.4byte BattleScript_EffectAttackerDefenseDownHit
 	.4byte BattleScript_EffectBodyPress
+	.4byte BattleScript_EffectExpandingForce
+	.4byte BattleScript_EffectScaleShot
+	.4byte BattleScript_EffectMeteorBeam
+	.4byte BattleScript_EffectRisingVoltage
+	.4byte BattleScript_EffectCorrosiveGas
+	.4byte BattleScript_EffectBeakBlast
+	.4byte BattleScript_EffectTerrainPulse
+	.4byte BattleScript_EffectCourtChange
+	.4byte BattleScript_EffectShellTrap
+
+BattleScript_EffectShellTrap::
+	checkshelltrap BS_ATTACKER, BattleScript_MoveEnd
+	clearshelltrap BS_ATTACKER
+	attackcanceler
+	attackstring
+	goto BattleScript_ButItFailed
```

Comment:
Does this mean Shell Trap is incomplete? It doesnt seem to trigger after laying the trap and getting hit, it always says But It Failed.